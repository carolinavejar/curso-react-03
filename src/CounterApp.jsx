import PropTypes from 'prop-types'


export const CounterApp = ( { 
    value
} )=> { 

    return (
    <>
        <h1>CounterApp</h1>
        <h2> { value } </h2>
        
        {/* <code>{ JSON.stringify(newMessage) }</code> */}
    </>);
};

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
