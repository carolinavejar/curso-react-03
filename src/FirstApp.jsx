import PropTypes from 'prop-types'


export const FirstApp = ( { 
    title, 
    subTitle,
    nameUser
} )=> { 

    return (
    <>
        <h1>Primera aplicación</h1>
        <h2> { title } </h2>
        <h3> { subTitle} </h3>
        <h3> { nameUser } </h3>
        
        {/* <code>{ JSON.stringify(newMessage) }</code> */}
    </>);
};

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number
}

FirstApp.defaultProps = {
    title: "Sin título",
    subTitle: "Sin subtítulo",
    nameUser: "Carito"
}