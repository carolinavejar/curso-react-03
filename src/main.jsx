import React from "react";
import ReactDOM from "react-dom/client";

import  { App as HelloWorldApp } from './HelloWorldApp'
import  { FirstApp } from './FirstApp';
import  './style.css'


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <FirstApp title="Titulo de Pruebas" subTitle={ 123 }/>
    </React.StrictMode>
)