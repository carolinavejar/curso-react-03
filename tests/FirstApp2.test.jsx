/**
* @jest-environment jsdom
*/

import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en First App', ()=> {
    const title = "Titulo ...";
    test('Debe hacer match con el snapshot', ()=> {
        const { container } = render(< FirstApp title= {title} />);

        expect (container).toMatchSnapshot();
    });

    test('Debe mostrar titulo: Titulo ...', ()=> {
        render(< FirstApp title= {title} />);

        expect (screen.getByText(title)).toBeTruthy();
    });

    test('Debe mostrar el titulo en un h1', ()=> {
        render(< FirstApp title= {title} />);

        expect (screen.getByRole('heading', 
        { level: 1}).innerHTML).toContain

    });

    test('Debe mostrar el subtitulo enviado por props', ()=> {
        const subtitle = "Subtitulo";
        render(
        < FirstApp title= {title} subTitle={subtitle} />);

        expect (screen.getAllByText(subtitle).length).toBe(1);

    });


})