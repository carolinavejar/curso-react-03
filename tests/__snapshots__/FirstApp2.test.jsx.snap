// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en First App Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Primera aplicación
  </h1>
  <h2
    data-testid="test-title"
  >
    Titulo ...
  </h2>
  <h3>
    Sin subtítulo
  </h3>
  <h3>
     
    Carito
     
  </h3>
</div>
`;
